staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -0,0 +1,174 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	authenticationv1alpha1 "k8s.io/api/authentication/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	authenticationv1alpha1client "k8s.io/client-go/kubernetes/typed/authentication/v1alpha1"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/templates"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/yaml"
+)
+
+// WhoAmIOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type WhoAmIOptions struct {
+	AuthClient authenticationv1alpha1client.AuthenticationV1alpha1Interface
+
+	TablePrinter printers.ResourcePrinter
+	OutputFormat *string
+
+	genericclioptions.IOStreams
+	PrintFlags *genericclioptions.PrintFlags
+}
+
+var (
+	whoAmILong = templates.LongDesc(`
+		Check who you are and your attributes (groups, extra).
+
+        This command is helpful to get yourself aware of the current user attributes, 
+        especially when dynamic authentication, e.g., token webhook, auth proxy, or OIDC provider, 
+        is enabled in the Kubernetes cluster.
+	`)
+
+	whoAmIExample = templates.Examples(`
+		# Get your subject attributes.
+		kubectl auth who-am-i
+		
+		# Get your subject attributes in JSON format.
+		kubectl auth who-am-i -o json
+	`)
+)
+
+// NewCmdWhoAmI returns an initialized Command for 'auth who-am-i' sub command
+func NewCmdWhoAmI(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &WhoAmIOptions{
+		OutputFormat: pointer.String(""),
+		IOStreams:    streams,
+		PrintFlags:   genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "who-am-i",
+		DisableFlagsInUseLine: true,
+		Short:                 "Check self subject attributes.",
+		Long:                  whoAmILong,
+		Example:               whoAmIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.RunAttributesGet())
+		},
+	}
+
+	allowedFormats := []string{"yaml", "json"}
+	cmd.Flags().StringVarP(o.OutputFormat, "output", "o", *o.OutputFormat, fmt.Sprintf(`Output format. One of: (%s).`, strings.Join(allowedFormats, ", ")))
```

Comment:
I was expecting a more "normal" printer that also handles templating, like what `kubectl get` uses.

Reply:
There is a problem that the printer you are thinking about only works with resources, but I'd rather like to output only `userInfo` which has neither kind nor metadata.

 To solve the issue, I decided to add a custom printer, but maybe there is a better way.

Reply:
How about injecting the kind and metadata so that it can be printed correctly? We do that already see https://github.com/kubernetes/kubernetes/blob/5a156349f6fdd31857249c30ebded7fcf19d30cf/staging/src/k8s.io/kubectl/pkg/cmd/create/create_job.go#L236

Reply:
I migrated to use printers from the library and manually tested go-template, jsonpath outputs.